t/unit/transport/test_redis.py

```diff
@@ -835,88 +953,86 @@ def test_rediss_connection(self):
 
-    def teardown(self):
+    def tearDown(self):
         self.connection.close()
+        self.queue = None
+        self.exchange = None
+        global _fake_redis_client
+        _fake_redis_client = None
 
-    @mock.replace_module_value(redis.redis, 'VERSION', [3, 0, 0])
-    def test_publish__get_redispyv3(self):
-        channel = self.connection.channel()
-        producer = Producer(channel, self.exchange, routing_key='test_Redis')
-        self.queue(channel).declare()
-
-        producer.publish({'hello': 'world'})
-
-        assert self.queue(channel).get().payload == {'hello': 'world'}
-        assert self.queue(channel).get() is None
-        assert self.queue(channel).get() is None
-        assert self.queue(channel).get() is None
-
-    @mock.replace_module_value(redis.redis, 'VERSION', [2, 5, 10])
-    def test_publish__get_redispyv2(self):
-        channel = self.connection.channel()
-        producer = Producer(channel, self.exchange, routing_key='test_Redis')
-        self.queue(channel).declare()
-
-        producer.publish({'hello': 'world'})
-
-        assert self.queue(channel).get().payload == {'hello': 'world'}
-        assert self.queue(channel).get() is None
-        assert self.queue(channel).get() is None
-        assert self.queue(channel).get() is None
-
-    def test_publish__consume(self):
-        connection = Connection(transport=Transport)
-        channel = connection.channel()
-        producer = Producer(channel, self.exchange, routing_key='test_Redis')
-        consumer = Consumer(channel, queues=[self.queue])
-
-        producer.publish({'hello2': 'world2'})
-        _received = []
-
-        def callback(message_data, message):
-            _received.append(message_data)
-            message.ack()
-
-        consumer.register_callback(callback)
-        consumer.consume()
-
-        assert channel in channel.connection.cycle._channels
-        try:
-            connection.drain_events(timeout=1)
-            assert _received
-            with pytest.raises(socket.timeout):
-                connection.drain_events(timeout=0.01)
-        finally:
-            channel.close()
+    def create_connection(self, **kwargs):
+        kwargs.setdefault('transport_options', {'fanout_patterns': True})
+        return Connection(transport=FakeRedisKombuTransport, **kwargs)
+
+    # @skip("This test is currently broken")
+    # def test_publish__get(self):
+    #     channel = self.connection.channel()
+    #     producer = Producer(channel, self.exchange, routing_key='test_Redis')
+    #     self.queue(channel).declare()
+    #
+    #     producer.publish({'hello': 'world'})
+    #
+    #     assert self.queue(channel).get().payload == {'hello': 'world'}
+    #     assert self.queue(channel).get() is None
+    #     assert self.queue(channel).get() is None
+    #     assert self.queue(channel).get() is None
+
+    # @skip("This test is currently broken")
```

Comment:
disabling these two tests should be removed? or kept for future improvement? don't they affect any current code?

Reply:
OK

Reply:
I've moved these tests down to a new test class that is based on mocks like they originally used.  Now the tests pass -- will see how the latest build goes, maybe a flake8 error to fix, maybe not.